{"cells":[{"cell_type":"code","execution_count":7,"metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":[" * Serving Flask app '__main__'\n"," * Debug mode: on\n"]},{"name":"stderr","output_type":"stream","text":["WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.\n"," * Running on http://127.0.0.1:5000\n","Press CTRL+C to quit\n"," * Restarting with stat\n","Traceback (most recent call last):\n","  File \"<frozen runpy>\", line 198, in _run_module_as_main\n","  File \"<frozen runpy>\", line 88, in _run_code\n","  File \"/home/codespace/.local/lib/python3.12/site-packages/ipykernel_launcher.py\", line 18, in <module>\n","    app.launch_new_instance()\n","  File \"/home/codespace/.local/lib/python3.12/site-packages/traitlets/config/application.py\", line 1074, in launch_instance\n","    app.initialize(argv)\n","  File \"/home/codespace/.local/lib/python3.12/site-packages/traitlets/config/application.py\", line 118, in inner\n","    return method(app, *args, **kwargs)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/home/codespace/.local/lib/python3.12/site-packages/ipykernel/kernelapp.py\", line 692, in initialize\n","    self.init_sockets()\n","  File \"/home/codespace/.local/lib/python3.12/site-packages/ipykernel/kernelapp.py\", line 331, in init_sockets\n","    self.shell_port = self._bind_socket(self.shell_socket, self.shell_port)\n","                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/home/codespace/.local/lib/python3.12/site-packages/ipykernel/kernelapp.py\", line 253, in _bind_socket\n","    return self._try_bind_socket(s, port)\n","           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n","  File \"/home/codespace/.local/lib/python3.12/site-packages/ipykernel/kernelapp.py\", line 229, in _try_bind_socket\n","    s.bind(\"tcp://%s:%i\" % (self.ip, port))\n","  File \"/home/codespace/.local/lib/python3.12/site-packages/zmq/sugar/socket.py\", line 311, in bind\n","    super().bind(addr)\n","  File \"_zmq.py\", line 917, in zmq.backend.cython._zmq.Socket.bind\n","  File \"_zmq.py\", line 179, in zmq.backend.cython._zmq._check_rc\n","zmq.error.ZMQError: Address already in use (addr='tcp://127.0.0.1:9007')\n"]},{"ename":"SystemExit","evalue":"1","output_type":"error","traceback":["An exception has occurred, use %tb to see the full traceback.\n","\u001b[0;31mSystemExit\u001b[0m\u001b[0;31m:\u001b[0m 1\n"]}],"source":["from flask import Flask\n","\n","app = Flask(__name__)\n","\n","@app.route('/')\n","def hello() :\n","    return \"Helloworld\"\n","\n","if __name__ == \"__main__\":\n","    app.run(debug = True)"]}],"metadata":{"kernelspec":{"display_name":"Python 3","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.12.1"}},"nbformat":4,"nbformat_minor":2}
